datasetinsights/datasets/transformers/base.py

```diff
@@ -0,0 +1,47 @@
+from abc import ABC, abstractmethod
+
+_registry = {}
+
+
+def get_dataset_transformer(format, **kwargs):
+    """
+    Returns instantiated transformer object based on the provided conversion
+    format from a registry.
+
+    Args:
+        format (str): Conversion format to be used for dataset transformation.
+
+    Returns: Transformer object instance.
+
+    """
+    if format in _registry.keys():
+        transformer = _registry[format]
+    else:
+        raise ValueError(
+            f"Transformer not found for conversion format '{format}'"
+        )
+
+    return transformer(**kwargs)
+
+
+class DatasetTransformer(ABC):
+    """ Base class for all ddataset transformer.
+    """
+
+    def __init__(self, **kwargs):
```

Comment:
Do we need the __init__ method here as part of the abstract class?

Reply:
Removed.